monkelib/API/get_data.py

```diff
@@ -0,0 +1,145 @@
+"""
+Functional description:
+    - This file contains the functions that are used to get data from the GitHub API.
+    - Helper functions are prefixed with an underscore, and they should not be used outside of this file.
+    - API model taken from model.py file.
+"""
+# --- Imports ---
+import requests
+import json
+import pandas as pd
+
+
+# --- Helper functions ---
+def _save_to_json(data, filename) -> None:
```

Comment:
What is the meaning of doing a _ before the function names?

Reply:
> What is the meaning of doing a _ before the function names?
I have no clue on how to demonstrate it for a specific part of the code but if you go into monkelib/API/get_data.py at the start of the file, there is a small """ functional requirements section """  that provides some information on what that underscore means. It just means that it is a helper function that shouldn't be called outside of the get_data.py file

Reply:
Ok, sounds good. But how do you decide wether a function should not be used outside of a module?

Reply:
> > What is the meaning of doing a _ before the function names?
> 
> I have no clue on how to demonstrate it for a specific part of the code but if you go into monkelib/API/get_data.py at the start of the file, there is a small """ functional requirements section """ that provides some information on what that underscore means. It just means that it is a helper function that shouldn't be called outside of the get_data.py file

Ok, sounds good. But how do you decide wether a function should not be used outside of a module?

Reply:
Will depend on what type of usage it will have, and that is subjective but for example in the case of the _load_secret() function, since we are only calling the API from that file, it makes sense for it to be a function that is meant to only used inside of that file